src/components/WeatherWidget/WeatherWidget.jsx

```diff
@@ -0,0 +1,132 @@
+import { StyledWrapper, StyledWrapperWeather } from "./WeatherWidget.styles";
+import { useState, useEffect } from "react";
+import GetLocation from "../../services/location";
+import GetWeather from "../../api/weather";
+import CircularProgress from "@mui/material/CircularProgress";
+import Box from "@mui/material/Box";
+import { Switch, FormControlLabel } from "@mui/material";
+
+const rainyImage = process.env.PUBLIC_URL + "/images/rainy-day.png";
+const sunnyImage = process.env.PUBLIC_URL + "/images/sun.png";
+const snowyImage = process.env.PUBLIC_URL + "/images/snow.png";
+const windImage = process.env.PUBLIC_URL + "/images/wind.png";
+const cloudImage = process.env.PUBLIC_URL + "/images/cloud.png";
+
+const weatherImages = {
+  rainy: {
+    condition: (data) => data.precipitation_probability > 50,
+    image: rainyImage,
+    description: "Rainy image",
+  },
+  snowy: {
+    condition: (data) => data.snowfall > 0,
+    image: snowyImage,
+    description: "Snowy image",
+  },
+  windy: {
+    condition: (data) => data.windspeed > 20,
+    image: windImage,
+    description: "Windy image",
+  },
+  sunny: {
+    condition: (data) => data.temperature > 20,
+    image: sunnyImage,
+    description: "Sunny image",
+  },
+  cloud: {
+    condition: () => true,
+    image: cloudImage,
+    description: "Cloudy image",
+  },
+};
+
+export const WeatherWidget = () => {
+  const [weatherData, setWeatherData] = useState(null);
+  const [error, setError] = useState(null);
+  const [temperatureUnit, setTemperatureUnit] = useState("C");
+
+  useEffect(() => {
+    const fetchWeather = async () => {
+      try {
+        const location = await GetLocation();
+        const weather = await GetWeather(location);
+        setWeatherData(weather);
+      } catch (err) {
+        setError(err.message);
+      }
+    };
+
+    fetchWeather();
+  }, []);
+
+  const getWeatherImage = () => {
+    if (!weatherData)
+      return { image: cloudImage, description: "Cloudy weather" };
+    for (const key in weatherImages) {
+      if (weatherImages[key].condition(weatherData)) {
+        return {
+          image: weatherImages[key].image,
+          description: weatherImages[key].description,
+        };
+      }
+    }
+  };
+
+  const weather = getWeatherImage();
```

Comment:
```suggestion
  const weatherImage = getWeatherImage();
```

Variable name to be always reflecting exactly what the variable contains. If it was named `weather` I'd expect this to hold weather data.